modules/host/update.go

```diff
@@ -313,7 +295,25 @@ func (h *Host) ProcessConsensusChange(cc modules.ConsensusChange) {
 		h.log.Println(err)
 	}
 
-	// Handle the list of action items.
+	// Handle the list of action items. Action items require host locks and
+	// potentially require waiting for long periods of time while various
+	// network communications finish. To prevent the host lock from being held
+	// while the network I/O is blocking the action item execution, the action
+	// items are handled in their own thread. A wait group is used to prevent
+	// the host from releasing ThreadGroup control until all threads that this
+	// function have spawned have completed.
+	wg := new(sync.WaitGroup)
+	// The host should not release thread group control until all spawned
+	// threads have completed execution, though waiting for them should not
+	// block this function. The host should also not release thread group
+	// control until this function has terminated, hence the use of 'defer
+	// func() { go func() }' to coordinate waiting and done-ing.
+	defer func() {
+		go func() {
+			wg.Wait()
```

Comment:
calling `wg.Wait` in a goroutine seems dangerous. It means that during resync, you'll be spawning `threadedHandleActionItem` goroutines at an unbounded rate. It's probably fine if you don't have tons of action items, but it makes me a little uneasy.
Would it be equivalent to simply call `tg.Add` in each `threadedHandleActionItem`, and remove the `WaitGroup`? Or is it important that each set of action items is processed atomically?